### PR TITLE
DRAFT: Readvalue() optimization

### PR DESCRIPTION
**Pull request by rofl0r**
this should speed up Readvalue() which is one of the most used functions during script executions (taking up as much as 3% cpu in my tests). inlining it *should* speed it up.
"should" because it's hard to measure any improvements to script interpreter code with the --fps method which is only adequate to measure gfx performance.

### CODE REVIEW

File: Engine/script/runtimescriptvalue.h [hunk 1]
```diff
@@ -309,13 +310,63 @@ struct RuntimeScriptValue
         return !(*this == rval);
     }
 
+    // FIXME: find out all certain cases when we are reading a pointer and store it
+    // as 32-bit value here. There should be a solution to distinct these cases and
+    // store value differently, otherwise it won't work for 64-bit build.
```
Comment (ivan-mogilko) [inline Engine/script/runtimescriptvalue.h]: Frankly, I don't remember what this comment means and whether it is still applicable.

**Comment by ivan-mogilko** [comment]
Sorry. the title sais "draft", but there were no ammendments for 20 days; this probably may be merged as it is then?

TBH I don't like having extra header for just 1 function, but the code is constantly reorganized as it is refactored and rewritten, so that may not be a big deal.

**Comment by rofl0r** [comment]
it means it's a proposed change, waiting for discussion. but yeah, you can go ahead and merge it, at the very least it untangles the recursive dependencies of headers. it's a bit hard to measure performance tweaks without a way to benchmark, unfortunately.

**Comment by sonneveld** [comment]
> it's a bit hard to measure performance tweaks without a way to benchmark, unfortunately

Do you use the opengl drivers? It should be fairly low overhead, especially with vsync disabled. You could create a game with a blank room and write test benchmark test scripts.

**Comment by rofl0r** [comment]
> Do you use the opengl drivers?

no, those are really slow on my device. as stated in #821, the only way to get reliable script interpreter performance measurements is by turning rendering off completely.

**Comment by sonneveld** [comment]
> as stated in #821, the only way to get reliable script interpreter performance measurements is by turning rendering off completely.

Sure, but until something like that is implemented, this is the best we've got.

**Pull request merged**